ecommerce_extensions/payment/views/edupay.py

```diff
@@ -0,0 +1,296 @@
+            billing_address = self._get_billing_address(edupay_response)
+
+            self.handle_order_placement(
+                order_number,
+                user,
+                basket,
+                None,
+                shipping_method,
+                shipping_charge,
+                billing_address,
+                order_total,
+                request=request,
+            )
+
+            return HttpResponse(status=status.HTTP_200_OK)
+        except:  # pylint: disable=bare-except
+            payment_processor = self.payment_processor.NAME.title() if self.payment_processor else None
+            logger.exception(self.order_placement_failure_msg, payment_processor, basket.id)
+            return HttpResponse(status=status.HTTP_200_OK)
+
+    def get(self, request, *args, **kwargs):
+        # pylint: disable=unused-argument
+        """Handle an incoming user returned to us by PP after processing payment."""
+        edupay_response = request.GET.dict()
+        try:
+            transaction_id = edupay_response.get('transactionId')
+            order_number = edupay_response.get('referenceCode')
+            basket_id = OrderNumberGenerator().basket_id(order_number)
+
+            logger.info(
+                'Received [%s] payer notification for transaction [%s], associated with basket [%d].',
+                EdnxPP.NAME,
+                transaction_id,
+                basket_id,
+            )
+
+            basket = self._get_basket(basket_id)
+
+            if not basket:
+                logger.error('Received payer notification for non-existent basket [%s].', basket_id)
+                return redirect(reverse('payment_error'))
+        except:  # pylint: disable=bare-except
+            return redirect(reverse('payment_error'))
+
+        receipt_url = get_receipt_page_url(
+            order_number=basket.order_number,
+            site_configuration=basket.site.siteconfiguration,
+        )
+
+        try:
+            return redirect(receipt_url)
+        except:  # pylint: disable=bare-except
+            payment_processor = self.payment_processor.NAME.title() if self.payment_processor else None
+            logger.exception(self.order_placement_failure_msg, payment_processor, basket.id)
+            return redirect(receipt_url)
+
+
+class EdnxPaymentStatusView(APIView):
+    """Polls payment status.
+
+    Return the current status of the payment by polling the order
+    or the basket status.
+    """
+    def get(self, request, order_number):
+        """Provide confirmation of payment."""
+        order_id = decode_string(order_number)
+        payment_status = self._get_payment_status(order_id)
+        if payment_status != 'complete':
+            return render(request, 'payment/pending_status.html', {'msg':payment_status})
+        else:
+            try:
+                order = Order.objects.get(number=order_id)
+            except:  # pylint: disable=bare-except
```

Comment:
This raises and specific exception, why are you catching everything

Reply:
What do you suggest instead? catch `Order.DoesNotExist` ?